### PR TITLE
MAT 6727 created excel key worksheet

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-6727](https://jira.cms.gov/browse/MAT-6727)
(Optional) Related Tickets:

### Summary
created excel key worksheet

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/controllers/export.controller.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export class ExportController {
+  constructor(private readonly exportService: ExportService) {}
+
+  @UseGuards(AuthGuard)
+  @Get()
+  @Header(
+    'Content-Type',
+    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+  )
+  @Header('Content-Disposition', 'attachment; filename="exampleWorkbook.xlsx"')
```
Comment (adongare) [inline src/controllers/export.controller.ts:16]: file name is going to be dynamic based on testcase. I'm not sure if you can customize with annotation. if not, you could move this inside the method and set the response header manually. that way you can generate filename dynamically e.g.

```
response.setHeader(
     'Content-Disposition',
     'attachment; filename=filename.xlsx',
   );
  
```

Comment (sb-prateekkeerthi) [reply src/controllers/export.controller.ts:16]: Yes, I do have code for making it dynamic but In my ticket I didn't know if I have to do that:

@Get(':filename')
@Header(
     'Content-Type',
     'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
   )
 @Header('Content-Disposition', 'attachment; filename*=UTF-8\'\'{filename}.xlsx')
async getFile(@Res() res: Response) {

Comment (adongare) [reply src/controllers/export.controller.ts:16]: gotcha, thanks!

**Review by adongare — APPROVED** [review]
